core/src/main/java/org/infinispan/atomic/AtomicMapLookup.java

```diff
@@ -142,6 +129,7 @@
     * @param <MK>  key param of the cache
     */
    public static <MK> void removeAtomicMap(Cache<MK, ?> cache, MK key) {
-      cache.getAdvancedCache().withFlags(Flag.IGNORE_RETURN_VALUES).remove(key);
+      // This removes any entry from the cache but includes special handling for fine-grained maps
+      FineGrainedAtomicMapProxyImpl.removeMap((Cache<Object, Object>) cache, key);
```

Comment:
Yes, for anything but fine grained it works just like a regular cache.remove(). Non-fine grained maps are just single entries, so no problem with that.